operator/src/test/java/oracle/kubernetes/operator/steps/ServerDownIteratorStepTest.java

```diff
@@ -0,0 +1,383 @@
+            .withSpec(createDomainSpec());
+  }
+
+  private DomainSpec createDomainSpec() {
+    return new DomainSpec()
+            .withDomainUid(UID)
+            .withWebLogicCredentialsSecret(new V1SecretReference().name(CREDENTIALS_SECRET_NAME))
+            .withIncludeServerOutInPodLog(INCLUDE_SERVER_OUT_IN_POD_LOG)
+            .withImage(LATEST_IMAGE);
+  }
+
+  private static void addServer(DomainPresenceInfo domainPresenceInfo, String serverName) {
+    if (serverName.equals(ADMIN)) {
+      domainPresenceInfo.setServerPod(serverName, createReadyPod(serverName));
+    } else {
+      domainPresenceInfo.setServerPod(serverName, createPod(serverName));
+    }
+  }
+
+  private static V1Pod createReadyPod(String serverName) {
+    return new V1Pod().metadata(withNames(new V1ObjectMeta().namespace(NS), serverName))
+            .spec(new V1PodSpec().nodeName("Node1"))
+            .status(new V1PodStatus().phase("Running")
+            .addConditionsItem(new V1PodCondition().type("Ready").status("True")));
+  }
+
+  private static V1Pod createPod(String serverName) {
+    return new V1Pod().metadata(withNames(new V1ObjectMeta().namespace(NS), serverName));
+  }
+
+  private static V1ObjectMeta withNames(V1ObjectMeta objectMeta, String serverName) {
+    return objectMeta
+            .name(LegalNames.toPodName(UID, serverName))
+            .putLabelsItem(LabelConstants.SERVERNAME_LABEL, serverName);
+  }
+
+  /**
+   * Setup env for tests.
+   * @throws NoSuchFieldException if TestStepFactory fails to install
+   */
+  @Before
+  public void setUp() throws NoSuchFieldException {
+    mementos.add(TestUtils.silenceOperatorLogger().ignoringLoggedExceptions(ApiException.class));
+    mementos.add(TuningParametersStub.install());
+    mementos.add(testSupport.install());
+
+    testSupport.defineResources(domain);
+    testSupport
+            .addToPacket(ProcessingConstants.DOMAIN_TOPOLOGY, domainConfig)
+            .addDomainPresenceInfo(domainPresenceInfo);
+    testSupport.addComponent(
+            ProcessingConstants.PODWATCHER_COMPONENT_NAME,
+            PodAwaiterStepFactory.class,
+            new PodHelperTestBase.DelayedPodAwaiterStepFactory(1));
+
+  }
+
+  /**
+   * Cleanup env after tests.
+   * @throws Exception if test support failed
+   */
+  @After
+  public void tearDown() throws Exception {
+    for (Memento memento : mementos) {
+      memento.revert();
+    }
+
+    testSupport.throwOnCompletionFailure();
+  }
+
+  @Test
+  public void withConcurrencyOf1_bothClusteredServersShutdownSequentially() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(1).withReplicas(1);
+    addWlsCluster(CLUSTER, 8001, MS1, MS2);
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
```

Comment:
This line that executes the code you're testing is repeated in each test, with the only real differences being that in some tests you create infos without a cluster, in some you create them with a cluster, and in one you create them for multiple clusters. As written now, it is hard at first glance to see what is going on.

This could be clarified in a number of ways. Maybe the cleanest would use a builder pattern, so that you could do something like.

```
createShutdownInfos()
    .forServers(MS1, MS2)
    .forClusteredServers(CLUSTER, MS3, MS4)
    .shutdown();
```

That would greatly improve readability of the tests. Also, please make sure to leave a blank line between the execution portion of each unit test and the assertions.